components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/state/OAuthRequestStateValidator.java

```diff
@@ -0,0 +1,178 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.endpoint.state;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.endpoint.OAuthRequestWrapper;
+import org.wso2.carbon.identity.oauth.endpoint.exception.AccessDeniedException;
+import org.wso2.carbon.identity.oauth.endpoint.exception.BadRequestException;
+import org.wso2.carbon.identity.oauth.endpoint.exception.InvalidApplicationClientException;
+import org.wso2.carbon.identity.oauth.endpoint.exception.InvalidRequestException;
+import org.wso2.carbon.identity.oauth.endpoint.exception.InvalidRequestParentException;
+import org.wso2.carbon.identity.oauth.endpoint.message.OAuthMessage;
+import org.wso2.carbon.identity.oauth.endpoint.util.EndpointUtil;
+
+import static org.wso2.carbon.identity.oauth.endpoint.state.OAuthAuthorizeState.AUTHENTICATION_RESPONSE;
+import static org.wso2.carbon.identity.oauth.endpoint.state.OAuthAuthorizeState.INITIAL_REQUEST;
+import static org.wso2.carbon.identity.oauth.endpoint.state.OAuthAuthorizeState.USER_CONSENT_RESPONSE;
+
+public class OAuthRequestStateValidator {
+
+    private static final Log log = LogFactory.getLog(OAuthRequestStateValidator.class);
+    private static final String REDIRECT_URI = "redirect_uri";
+
+    public OAuthAuthorizeState validateAndGetState(OAuthMessage oAuthMessage) throws InvalidRequestParentException {
+
+        if (handleToCommonauthState(oAuthMessage)) {
+            return OAuthAuthorizeState.PASSTHROUGH_TO_COMMONAUTH;
+        }
+
+        validateRequest(oAuthMessage);
+
+        if (oAuthMessage.isInitialRequest()) {
+            validateInputParameters(oAuthMessage);
+            return INITIAL_REQUEST;
+        } else if (oAuthMessage.isAuthResponseFromFramework()) {
+            return AUTHENTICATION_RESPONSE;
+        } else if (oAuthMessage.isConsentResponseFromUser()) {
+            return USER_CONSENT_RESPONSE;
+        } else {
+            return handleInvalidRequest();
+        }
+    }
+
+    private OAuthAuthorizeState handleInvalidRequest() throws InvalidRequestException {
+        // Invalid request
+        if (log.isDebugEnabled()) {
+            log.debug("Invalid authorization request");
+        }
+
+        throw new InvalidRequestException("Invalid authorization request");
+    }
+
+    private boolean handleToCommonauthState(OAuthMessage oAuthMessage) {
+        if (oAuthMessage.isRequestToCommonauth() && oAuthMessage.getFlowStatus() == null) {
```

Comment:
Can't we simplify the login to 
`return oAuthMessage.isRequestToCommonauth() && oAuthMessage.getFlowStatus() == null`

Reply:
done